### ISSUE TITLE
Bool to Float - PyTorch 2.12 - FL_Inpaint_Crop.py

### ISSUE THREAD

**Issue by IrradiatedHaggis** [filliptm/ComfyUI_Fill-Nodes#106]
PyTorch 2.12 no longer works with the prior version of FL_Inpaint_Crop.py due to how it handles F.conv2d - 2.12 drops bool support - it needs to be a float instead.

This should fix it:

```python
    # GPU-accelerated version of binary_fill_holes
    def binary_fill_holes_gpu(self, binary_mask_tensor):
        # Create a kernel for morphological operations
        kernel_size = 3
        kernel = torch.ones(1, 1, kernel_size, kernel_size, device=binary_mask_tensor.device)

        # Perform closing operation (dilation followed by erosion)
        padding = kernel_size // 2
        padded = F.pad(binary_mask_tensor.unsqueeze(0).unsqueeze(0), (padding, padding, padding, padding), mode='constant', value=1)

        # Dilate - FIX: Cast padded to float() here
        dilated = F.conv2d(padded.float(), kernel, padding=0)
        dilated = (dilated > 0).float()

        # Erode
        inverted = 1 - dilated
        eroded_inv = F.conv2d(inverted, kernel, padding=0)
        closed = 1 - (eroded_inv > 0).float()

        # Remove padding
        closed = closed[0, 0, padding:-padding, padding:-padding]

        # Fill holes (find isolated regions of 0s surrounded by 1s)
        h, w = closed.shape
        filled = closed.clone()

        # Create a mask of border pixels
        border = torch.zeros_like(filled)
        border[0, :] = 1
        border[-1, :] = 1
        border[:, 0] = 1
        border[:, -1] = 1

        # Set border pixels to 0 in our working mask if they're 0 in the original
        seed = torch.ones_like(filled)
        seed[border == 1] = closed[border == 1]

        # Iteratively propagate the seed inward
        for _ in range(max(h, w)):
            # Pad for convolution
            padded_seed = F.pad(seed.unsqueeze(0).unsqueeze(0), (1, 1, 1, 1), mode='constant', value=0)

            # Use convolution to check neighbors
            neighbors = F.conv2d(padded_seed, torch.ones(1, 1, 3, 3, device=seed.device), padding=0)
            neighbors = neighbors[0, 0]

            # Propagate: if any neighbor is marked and the current pixel is not a foreground pixel in the original
            new_seed = (neighbors > 0).float() * (1 - closed)

            # Update seed with newly marked pixels
            seed = torch.clamp(seed + new_seed, 0, 1)

            # Check if we've converged
            if not new_seed.any():
                break

            # The holes are the areas that weren't reached
            holes = (1 - seed) * (1 - closed)

        # Add the holes to the original mask
        filled = torch.clamp(closed + holes, 0, 1)

        return filled
```

**Comment by filliptm** [comment]
Fixed by PR #108 plus follow-up commit 3078ffa. The GPU hole-fill path now casts bool masks before conv2d and preserves the original mask dimensions through the morphology operations.
